bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandlerFactory.java

```diff
@@ -67,17 +66,13 @@ protected void activate(ComponentContext componentContext) {
     }
 
     /**
-     * Disposes the {@link BaseThingHandlerFactory}. If this method is
-     * overridden by a sub class, the implementing method must call <code>super.deactivate(componentContext)</code>
-     * first.
+     * Disposes the {@link BaseThingHandlerFactory}. If this method is overridden by a sub class, the implementing
+     * method must call <code>super.deactivate(componentContext)</code> first.
      *
      * @param componentContext
      *            component context (must not be null)
      */
     protected void deactivate(ComponentContext componentContext) {
-        for (ServiceRegistration<ThingHandler> serviceRegistration : this.thingHandlers.values()) {
```

Comment:
This makes the implicit assumption that a ThingHandlerFactory does not have any dynamic dependencies, but that it is deactivated on any change of its dependent services. I think this must be clearly mentioned in the documentation - it is probably best if you could add a new section that explains how to deal with service dependencies in handlers (i.e. that they do not have a bundleContext anymore and thus should not themselves deal with their dependencies, but only get them reached in from the factory through their constructor).